.adr/0003-consensus-ledger-channels.md

```diff
@@ -0,0 +1,143 @@
+# 0003 -- Consensus ledger channel updates
+
+## Status
+
+Accepted
+
+## Definitions
+
+**L(x,y)** is the ledger channel between x and y.
+
+## Context
+
+A running ledger channel must serve concurrent funding requests originating from many different particpants. In the below channel network:
+
+```
+Alice ----.              .-- Doris
+           \            /
+Bob   ---- Irene --- Ivy --- Eric
+           /           \
+Charlie --/             \--- Frannie
+```
+
+Any client participant (A-F) can initate a virtualfund protocol with any other participant at any time, which will require updates to each ledger channel on the path between them. Updates to ledger channels must be ordered in some capacity so that (for example) the channel is not "overdrawn".
+
+The hub-hub channel L(Irene, Ivy) is particularly challenging, since each of Irene & Ivy can recieve a request that will require updates to L(Irene, Ivy) at any time, from any participant.
+
+
+## Considered Options
+
+### Complete Consensus Ledger Update
+
+Consensus channel runs the null-app. State updates must be by unanimous consensus (ie, signed by all participants).
+
+pros:
+- security guarantees as strong as the base protocol
+- no bespoke on-chain application logic
+- lower off-chain implementation complexity
+
+cons:
+- sacrificing some performance optimizations
+
+### Async Ledger Updates
+
+- https://www.notion.so/statechannels/RFC-13-Async-virtual-funding-b2b6ed9e39b34a7fbd362026dc248b0f
```

Comment:
It might be nice to just embed some of the content of the notion doc in this ADR so people can understand what async virtual funding is without having to access a private(I think?) notion doc.